### ISSUE TITLE
Update dependencies and peer dependencies

### ISSUE THREAD

**Issue by joebochill** [brightlayer-ui/react-native-component-library#140]
#### Describe the desired behavior
Check the current/latest version of all dependencies and peer dependencies. Update the package.json to include the latest.

#### Describe the current behavior
Using old versions of dependencies.

#### Is this request related to a current issue?
The newest version of NPM fails on npm install when there are peer dependency issues. 

#### Additional Context
After updating the versions of dependencies, make sure that things still work as expected.

**Comment by joebochill** [comment]
This might already be fixed by some of the recent updates

**Comment by huayunh** [comment]
Looking in to this — should be easy. Might already be fixed

**Comment by joebochill** [comment]
I went through and tried to make the updates all in one pass and ran into a bunch of errors at build time. I'm disinclined to try to make these updates so close to publishing, so I'm going to take off the high priority label and we can come back in and update these methodically later.

Storybook updates can probably be handled next quarter with the move to storybook 6.
Components need to be updated to the latest RN and jest, but I ran into some issues with the jest config when I updated to the latest versions of everything all at once.
